Gems/ROS2/Code/Source/Camera/CameraSensor.cpp

```diff
@@ -196,7 +195,8 @@ namespace ROS2
     {
         RequestFrame(
             cameraPose,
-            [header, publisher](const AZ::RPI::AttachmentReadback::ReadbackResult& result)
+            [header, publisher, &postProcess = std::as_const(m_cameraSensorDescription.m_cameraConfiguration.imagePostProcessingFunction)](
```

Comment:
Using &postProcess results in segfault while ending simulation. 
It seems that CameraSensor ends lifetime before lambda in certain scenarios 
Solution:
postProcess = m_cameraSensorDescription.m_cameraConfiguration.imagePostProcessingFunction

Reply:
Agree, great you catch it in the debugger @MateuszWasilewski.  I would consider using a shared pointer to function to omit copying. 
The last RequestFrame occurs when a lot of assets are destroyed (e.g. it is common to see in ROS the last frame with funny textures). Please mechanism that won't give undefined behavior here.

Reply:
The design is changed, please review the new one